### DISCUSSION TITLE
Issue with Syntax Highlighting and File Icons Disappearing after Extension Installation

### DISCUSSION THREAD

**Discussion by johnny-kc-wu** [coder/code-server#6391] [Q&A]
I've set up code-server as a Docker container on my Synology NAS, using Tailscale for VPN and reverse proxy. Initially, everything worked well. However, after adding some extensions and reloading the browser, the syntax highlighting and file icons vanished. This even caused the IDE to struggle with identifying document types.

Here's a snippet from my `docker-compose.yaml`:
```yal
services:
  vscs-johnny:
    image: codercom/code-server:latest
    container_name: vscode-johnny
    restart: always
    expose:
      - 8080
      - 8887
      - 8888
    ports:
      - 16880:8080
      - 8887:8887
      - 8888:8888
    user: 1026:100
    environment:
      - TZ=Asia/Japan
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /volume1/homes/user123:/home/coder/.config:rw
      - /volume1/homes/user123/projects:/home/coder/project:rw
```
the IDE:
![Screenshot 2023-08-19 145635](https://github.com/coder/code-server/assets/7566194/cd712972-791d-4c78-abcb-19a7ad39831d)
![Screenshot 2023-08-19 145153](https://github.com/coder/code-server/assets/7566194/ab3021f5-2687-43ac-9317-3a3b14aee9bc)

the part of code-server log:
```bash
[14:57:17] Getting Manifest... pkief.material-icon-theme
[14:57:20] Installing extension: pkief.material-icon-theme
[14:57:22] Extension signature verification is not done: pkief.material-icon-theme
[14:57:22] Extension installed successfully: pkief.material-icon-theme
[14:59:05] [172.19.0.1][41c8389e][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[14:59:05] [172.19.0.1][62b34b17][ExtensionHostConnection] <29592> Extension Host Process exited with code: 0, signal: null.
File not found: /usr/lib/code-server/lib/vscode/out/vsda_bg.wasm
File not found: /usr/lib/code-server/lib/vscode/out/vsda.js
[14:59:06] [172.19.0.1][1af5d66b][ManagementConnection] New connection established.
[14:59:07] [172.19.0.1][8fef1490][ExtensionHostConnection] New connection established.
[14:59:07] [172.19.0.1][8fef1490][ExtensionHostConnection] <31079> Launched Extension Host Process.
[IPC Library: Pty Host]  INFO Expanding terminal instance, old id 1 -> new id undefined
[IPC Library: Pty Host]  WARN Persistent process "1": Process had no disconnect runners but was an orphan
[IPC Library: Pty Host]  INFO Persistent process reconnection "1"
[IPC Library: Pty Host]  INFO Persistent process "1": Replaying 450 chars and 1 size events
[14:59:31] [172.19.0.1][e4f7af07][ExtensionHostConnection] <27284> Extension Host Process exited with code: 0, signal: null.
[15:00:05] [172.19.0.1][1af5d66b][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
```
![before_reload](https://github.com/coder/code-server/assets/7566194/97ce4c3d-1425-494b-ada2-710a0ac2319d)
![after_reload](https://github.com/coder/code-server/assets/7566194/4568014d-e8d5-406d-b5f6-f73c364ec1be)
After reloading the browser, the extensions that were previously installed and visible in the browser section disappeared. What caused this sudden disappearance, and what is the reason behind it?

![networking](https://github.com/coder/code-server/assets/7566194/2b3c742f-37d3-43b7-8624-e9f0074cf001)
Also, I noticed in the browser's networking log that `vsda_bg.wasm` and `vsda.js` were not found. What is the purpose of these files and why they might be triggering a `404` error?

**Comment by code-asher** [comment]
vsda seems to have something to do with signing.  It does not 
appear to exist at all in the OSS build of VS Code so I assume 
Microsoft adds it afterward to their non-OSS build.  From a brief 
look I think it is optional, though.

When I install the Material icon theme it shows up in the remote 
section (not the browser section) and when I reload the page it 
remains in the remote section so I am not sure why the behavior is 
different for you.
 
I installed 4.16.1 using the installation script and ran it 
directly with the default settings except with auth disabled.  I 
installed the extension through the UI.
